src/module.c

```diff
@@ -975,7 +1008,8 @@ int TSDB_delete(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 
     Series *series;
     RedisModuleKey *key;
-    const int status = GetSeries(ctx, argv[1], &key, &series, REDISMODULE_READ | REDISMODULE_WRITE);
+    const int status =
+        GetSeries(ctx, argv[1], &key, &series, REDISMODULE_READ | REDISMODULE_WRITE, false, false);
```

Comment:
@OfirMos you want to cleanup the rules here because it might upsert to the compacted keys:

```suggestion
        GetSeries(ctx, argv[1], &key, &series, REDISMODULE_READ | REDISMODULE_WRITE, true, false);
```

Reply:
@danni-m What you are suggesting isn't efficient, I'm calling deleteReferenceToDeletedSeries only when I see that there are rules for the series.